src/color.js

```diff
@@ -366,34 +365,40 @@ function translateColor (ycc, luma) {
 
   let intersection = null;
 
-  for (let i = 0; i < cubeFaces.length; i++) {
-    let cubeFace = cubeFaces[i];
-    intersection = findIntersection(line, cubeFace);
+  let cubeFaces = (lighter ? YCC_CUBE_FACES_WHITE : YCC_CUBE_FACES_BLACK);
+  for (let cubeFace of cubeFaces) {
+    let candidateIntersection = findIntersection(line, cubeFace);
+    if (candidateIntersection.z < 0 || candidateIntersection.z > 1)
+      continue;
```

Comment:
This section is the crux of the change.

Have a look at these two slides: 
https://docs.google.com/presentation/d/1wNR8g-mhAcbU_g-5S4FJ749b4Tn5KhIw3zA-RFunylo/edit#slide=id.g63d390087_1_363
https://docs.google.com/presentation/d/1wNR8g-mhAcbU_g-5S4FJ749b4Tn5KhIw3zA-RFunylo/edit#slide=id.g63d390087_1_383

When changing the luma, it's easy for the resulting coordinates to fall outside the cube. That would result in a "colour" with a value for `r`, `g` or `b` which is not within `[0, 255]`, i.e. not an actual colour. So, we need to clip the colour to the nearest side of the cube.

We do this by taking the three planes representing the three sides of the cube in the direction from the original point that we want to travel, i.e. lighter (towards white, which is `(0, 0, 1)`) or darker (towards black, `(0, 0, 0)`), and finding the intersection between the vertical line containing the original point and the nearest plane.

Originally, I assumed that the first intersection we found was ok. However, I forgot to take into account that - of course - those planes continue out past the edges of the cube.

So, the bug you found was from finding an intersection with a plane from the *wrong side* of the cube, which wasn't actually *on* the cube, simply because of the order in which we were testing for intersections. Hence, we were ending up with a negative value for `r`.

This change changes it so that we're checking *all three* potential sides, and making sure we pick the intersection which is closest to the original point.

So now this line, instead of breaking out of the loop, simply checks whether this is a *plausible* intersection (i.e. has a `z`/`luma` value which falls within the range of the cube).

Then, lines 381-382 ensure that we find the *right* intersection.